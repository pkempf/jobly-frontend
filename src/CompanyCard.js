import React from "react";
import Card from "react-bootstrap/Card";
import { Link } from "react-router-dom";
import "./CompanyCard.css";

const CompanyCard = ({ company }) => {
  return (
    <Link to={`/companies/${company.handle}`} className="CompanyCard">
      <Card className="mb-2">
        <Card.Body>
          <Card.Title>{company.name}</Card.Title>
          <Card.Text>{company.description}</Card.Text>
        </Card.Body>
      </Card>
    </Link>
  );
};

export default CompanyCard;
