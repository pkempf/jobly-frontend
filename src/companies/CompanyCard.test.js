import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CompanyCard from "./CompanyCard";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <CompanyCard company={{}} />
    </MemoryRouter>
  );
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard company={{}} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
