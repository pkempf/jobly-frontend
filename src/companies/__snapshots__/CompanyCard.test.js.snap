// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard"
    href="/companies/undefined"
  >
    <div
      class="mb-2 card"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        />
        <p
          class="card-text"
        />
      </div>
    </div>
  </a>
</DocumentFragment>
`;
